src/game/logic/object/experiencetracker.cpp

```diff
@@ -0,0 +1,181 @@
+    return true;
+}
+
+// zh: 0x0061B8F0 wb: 0x008A37E4
+bool ExperienceTracker::Can_Gain_Exp_For_Level(int32_t levelsToGain) const
+{
+    return m_currentLevel != std::min<VeterancyLevel>(m_currentLevel + levelsToGain, VeterancyLevel::VETERANCY_COUNT - 1);
+}
+
+// zh: 0x0061B910 wb: 0x008A381C
+void ExperienceTracker::Add_Experience_Points(int32_t experienceGain, bool applyMultiplier)
+{
+    if (m_experienceSink != ObjectID::OBJECT_UNK) {
+        auto *sink_object = g_theGameLogic->Find_Object_By_ID(m_experienceSink);
+        if (sink_object != nullptr) {
+            sink_object->Get_Experience_Tracker()->Add_Experience_Points(experienceGain * m_experienceMultiplier, applyMultiplier);
+            return;
+        }
+    }
+
+    if (!Is_Trainable()) {
+        return;
+    }
+
+    const auto old_level = m_currentLevel;
+    if (applyMultiplier) {
+        experienceGain *= m_experienceMultiplier;
+    }
+
+    m_currentExperience += experienceGain;
+    VeterancyLevel new_level = VeterancyLevel::VETERANCY_REGULAR;
+    // Loop until the penultimate VeterancyLevel as it reaches saturation
+    for (; new_level < VeterancyLevel::VETERANCY_COUNT - 1; ++new_level) {
+        if (m_currentExperience < m_parent->Get_Template()->Get_Experience_Required(new_level)) {
+            break;
+        }
+    }
+
+    m_currentLevel = new_level;
+    if (old_level != m_currentLevel) {
+        m_parent->On_Veterancy_Level_Changed(old_level, m_currentLevel, true);
+    }
+}
+
+// zh: 0x0061B9D0 wb: 0x008A3934
+void ExperienceTracker::Set_Experience_And_Level(int32_t experienceGain, bool unk)
+{
+    if (m_experienceSink != ObjectID::OBJECT_UNK) {
+        auto *sink_object = g_theGameLogic->Find_Object_By_ID(m_experienceSink);
+        if (sink_object != nullptr) {
+            sink_object->Get_Experience_Tracker()->Set_Experience_And_Level(experienceGain, unk);
+            return;
+        }
+    }
+
+    if (!Is_Trainable()) {
+        return;
+    }
+
+    const auto old_level = m_currentLevel;
+
+    m_currentExperience = experienceGain;
+
+    VeterancyLevel new_level = VeterancyLevel::VETERANCY_REGULAR;
+    // Loop until the penultimate VeterancyLevel as it reaches saturation
+    for (; new_level < VeterancyLevel::VETERANCY_COUNT - 1; ++new_level) {
+        if (m_currentExperience < m_parent->Get_Template()->Get_Experience_Required(new_level)) {
+            break;
+        }
+    }
+
+    m_currentLevel = new_level;
+    if (old_level != m_currentLevel) {
+        m_parent->On_Veterancy_Level_Changed(old_level, m_currentLevel, unk);
+    }
+}
+
+// zh: 0x0061BA70 wb: 0x008A3A1B
+void ExperienceTracker::CRC_Snapshot(Xfer *xfer)
+{
+    xfer->xferInt(&m_currentExperience);
+    xfer->xferUser(&m_currentLevel, sizeof(m_currentLevel));
+}
+
+// zh: 0x0061BAA0 wb: 0x008A3A51
+void ExperienceTracker::Xfer_Snapshot(Xfer *xfer)
+{
+    uint8_t version = 1;
+    xfer->xferVersion(&version, 1);
```

Comment:
`sizeof(version)`

Reply:
The 1 represents the version that it should be https://github.com/TheAssemblyArmada/Thyme/blob/84e604c484f5fbd890a570a40ad4593eb1feddc1/src/game/common/system/xfer.cpp#L32

You have to remember there are two callers for this function save and load. Under save the 1st parameter, version, is set to 1 and serialised. Under load the 1st parameter, version, is deserialised and verified to match the second parameter.

Reply:
Hmm ok. I wasn't sure what I am looking at here, because next line passes size.

Reply:
Yeah xfer functions are a bit weird. If you look at a few other classes xfers you'll see the general pattern.